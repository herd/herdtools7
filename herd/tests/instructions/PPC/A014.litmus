PPC A014
(* Test ROT32 vs ROT64 *)
{
0:r3=0xaaaaaaaaaaaaaaaa;
0:r5=0xaaaaaaaaaaaaaaaa;
0:r6=0xaaaaaaaaaaaaaaaa;
}

P0;
li r4,-1                 ;
rlwimi r3, r4, 4, 30, 31 ;
rlwimi r5, r4, 0, 30, 31 ;
rlwimi r6, r4, 16, 0, 15 ;
locations [0:r3; 0:r5; 0:r6;]


