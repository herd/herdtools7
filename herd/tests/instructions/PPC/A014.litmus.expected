Test A014 Required
States 1
0:r3=-1431655765; 0:r5=-1431655765; 0:r6=-21846;
Ok
Witnesses
Positive: 1 Negative: 0
Condition forall (true)
Observation A014 Always 1 0
Hash=f5fd1f3a6656a0390699cb32c0ad1af0

